### PR TITLE
Quickstart redesign

### PR DESCRIPTION
**Pull request by seanmooretechwriter**
Prepare new branch for redesign work.

### CODE REVIEW

**Comment by seanmooretechwriter** [comment]
PR linked to https://github.com/streamingfast/substreams/issues/110

**Comment by seanmooretechwriter** [comment]
Initial Quickstart redesign ready for review.

File: docs/getting-started/quickstart.md [hunk 1]
```diff
@@ -39,49 +43,69 @@ The [`run`](https://substreams.streamingfast.io/reference-and-specs/command-line
...
+- Use the [`substreams` CLI](https://substreams.streamingfast.io/reference-and-specs/command-line-interface) to run the module
 
-### **Basics**
+### Prerequisites
 
-The most basic approach to use Substreams is through the [`substreams` CLI](../reference-and-specs/command-line-interface.md), passing an endpoint and the name of the Rust function, or module handler, used by the compute engine for processing.&#x20;
+- Go compiler
+- Rust compiler
+- [`substreams` CLI](https://substreams.streamingfast.io/reference-and-specs/command-line-interface)
+- StreamingFast authentication key
```
Comment (maoueh) [inline docs/getting-started/quickstart.md:62]: Most of this should have been done by that point already. For example, there is `Run your first Substreams` higher, so I pretty sure that authentication and Substreams CLI has been dealt with.

The Go compiler, is definitely not required, what made you think it's required?

I would get rid of the `Prerequisites` section and just ensure prior installation step are covered.

File: docs/getting-started/quickstart.md [hunk 2]
```diff
@@ -39,49 +43,69 @@ The [`run`](https://substreams.streamingfast.io/reference-and-specs/command-line
...
+### 1. Create Substreams manifest
 
-The commands demonstrate how Substreams modules work by using the selected blockchain and instructing the compute engine to execute the `map_basic_eth` or `map_basic_sol` Rust functions, also called "_module handlers_".
+To create a "Substreams module", you must first create the manifest file. Use the example manifest provided in the [`substreams-ethereum-tutorial`](https://github.com/streamingfast/substreams-ethereum-tutorial)repository on GitHub as a guide. This example manifest includes the minimal required fields to demonstrate the core values that you must provide.
 
-Use the [`run`](https://substreams.streamingfast.io/reference-and-specs/command-line-interface#run) command for the Ethereum example by using:
+To use the example manifest, copy and paste it into a new file named substreams.yaml. Save this file in the root directory of your "Substreams module." You can find the example manifest in the [official GitHub repository for `substreams-ethereum-tutorial`](https://github.com/streamingfast/substreams-ethereum-tutorial).
```
Comment (maoueh) [inline docs/getting-started/quickstart.md:68]: https://github.com/streamingfast/substreams-ethereum-tutorial is reference twice. I would remove:

> Use the example manifest provided in the [`substreams-ethereum-tutorial`](https://github.com/streamingfast/substreams-ethereum-tutorial)repository on GitHub as a guide.

From the first paragraph. Then, both can be merged together in one.

Also, shouldn't `substreams.yaml` be in monospace so wrapped with backticks?

File: docs/getting-started/quickstart.md [hunk 3]
```diff
@@ -39,49 +43,69 @@ The [`run`](https://substreams.streamingfast.io/reference-and-specs/command-line
...
+    kind: map
+    inputs:
+      - source: sf.ethereum.type.v2.Block
+    output:
+      type: proto:eth.basicexample.v1.BasicExampleProtoData
 ```
-{% endcode %}
 
-Different blockchains have specific requirements for their data definitions. The code in the module handlers needs to be updated to match the expectations of the separate blockchains.
+### 2. Create Rust configuration file
```
Comment (maoueh) [inline docs/getting-started/quickstart.md:99]:
```suggestion
### 2. Create Rust manifest file
```

File: docs/getting-started/quickstart.md [hunk 4]
```diff
@@ -39,49 +43,69 @@ The [`run`](https://substreams.streamingfast.io/reference-and-specs/command-line
...
+
+To complete your new Substreams module, you must also create a Rust configuration file. Use the example configuration file from the [`substreams-ethereum-tutorial`](https://github.com/streamingfast/substreams-ethereum-tutorial) as a guide.
```
Comment (maoueh) [inline docs/getting-started/quickstart.md:101]: > Use the example configuration file from the [`substreams-ethereum-tutorial`](https://github.com/streamingfast/substreams-ethereum-tutorial) as a guide

This has been said already, let's ensure we say once, only in the first step, that `the full working tutorial is available at ... for reference blabla`.

File: docs/getting-started/quickstart.md [hunk 5]
```diff
@@ -93,157 +117,89 @@ crate-type = ["cdylib"]
 
 [dependencies]
 substreams = "0.5.0"
+substreams-ethereum = "0.8.0"
 prost = "0.11"
+num-bigint = "0.4"
```
Comment (maoueh) [inline docs/getting-started/quickstart.md:122]: I don't think this is one is strictly required for the tutorial

File: docs/getting-started/quickstart.md [hunk 6]
```diff
@@ -93,157 +117,89 @@ crate-type = ["cdylib"]
...
-{% endhint %}
+Use the `substreams protogen` command to generate the Rust code to communicate with the protobuf.
 
-## Quickstart examples overview
+```bash
+substreams protogen substreams-ethereum-tutorial.yaml
+```
 
-The relationships for the flow of data are defined in the Substreams manifest. Further information is available in the documentation for [defining complex data strategies](../reference-and-specs/manifests.md) through manifest files.
+### 4. Create Rust library and module handlers
```
Comment (maoueh) [inline docs/getting-started/quickstart.md:152]:
```suggestion
### 4. Create Substreams module handlers
```

File: docs/getting-started/quickstart.md [hunk 7]
```diff
@@ -93,157 +117,89 @@ crate-type = ["cdylib"]
...
 
-### Ethereum example
+Your Substreams module must contain a Rust library that houses the "module handlers." These handlers are responsible for handling blockchain data injected into the module at runtime. The example module handler extracts the version from the <i>Block</i> object in the `block.ver` property, logs it to the terminal, and assigns it to the `version` field of the [`BasicExampleProtoData`](https://github.com/streamingfast/substreams-ethereum-tutorial/blob/main/proto/basicexample.proto) protobuf.
```
Comment (maoueh) [inline docs/getting-started/quickstart.md:154]: We should change the actual tutorial to output block's hash and block number. The `ver` should never be mentioned, it's kind of a detail of the version of the block, but not all our chains have a `ver`. While `hash` and `number` will work for any blockchain.

Can you make the necessary change in the tutorial code as well as in this document.

File: docs/getting-started/quickstart.md [hunk 8]
```diff
@@ -93,157 +117,89 @@ crate-type = ["cdylib"]
...
-**Note**: The module handler in the [`lib.rs`](https://github.com/streamingfast/substreams-ethereum-tutorial/blob/main/src/lib.rs) file for the Ethereum example has code specific to the blockchain. The basic block structure for Ethereum blocks can be viewed in the code excerpt.
-{% endhint %}
+use substreams::{log};
+use substreams_ethereum::{pb as ethpb};
 
-{% code title="src/lib.rs" overflow="wrap" %}
-```rust
 #[substreams::handlers::map]
 fn map_basic_eth(block: ethpb::eth::v2::Block) -> Result<basicexample::BasicExampleProtoData, substreams::errors::Error> {
     log::info!("block.ver: {:#?}", block.ver);
```
Comment (maoueh) [inline docs/getting-started/quickstart.md:170]: Let's ensure also that only one log line is printed, the printed values should the block's hash/number

File: docs/getting-started/quickstart.md (hunk 2 above)
Comment (maoueh) [reply docs/getting-started/quickstart.md:68]: Is this a typo `... of your "Substreams module."` ? The ending dot within the `"`

File: docs/getting-started/quickstart.md [hunk 9]
```diff
@@ -93,157 +117,89 @@ crate-type = ["cdylib"]
...
-    log::info!("block.previous_blockhash: {:#?}", block.previous_blockhash);
-    log::info!("block.blockhash: {:#?}", block.blockhash);
-    log::info!("block.slot: {:#?}", block.slot);
-    Ok(basicexample::BasicExampleProtoData {blockhash: block.blockhash})
-}
-```
 {% endcode %}
 
-Follow the same steps [used for the Ethereum example](quickstart.md#ethereum-example) to [`run`](https://substreams.streamingfast.io/reference-and-specs/command-line-interface#run) the Solana example.
+To ensure that the [`run`](https://substreams.streamingfast.io/reference-and-specs/command-line-interface#run) command is executed correctly, you need to pass the proper endpoint, manifest name, and module handler name. The `--start-block` and `--stop-block` flags are optional, but they can help limit the results that are returned to the client.
```
Comment (maoueh) [inline docs/getting-started/quickstart.md:196]: Feels like it's not important for the quick start, should probably be moved in `Next Steps` for those you would maybe want to test a Substreams for another chain.

**Review by maoueh — CHANGES_REQUESTED** [review]

File: docs/getting-started/quickstart.md (hunk 1 above)
Comment (seanmooretechwriter) [reply docs/getting-started/quickstart.md:62]: Removed. Was matching the Google QSG here.

File: docs/getting-started/quickstart.md (hunk 2 above)
Comment (seanmooretechwriter) [reply docs/getting-started/quickstart.md:68]: Sentence removed. Content merged. No, that was correctly formatted, but I went ahead and removed the quotes for the second reference to Substreams module. The punctuation goes inside the quotes at the end of sentences "like this."

File: docs/getting-started/quickstart.md (hunk 3 above)
Comment (seanmooretechwriter) [reply docs/getting-started/quickstart.md:99]: Updated. 

(All the literature I've seen refers to this as a Rust configuration file.)

File: docs/getting-started/quickstart.md (hunk 4 above)
Comment (seanmooretechwriter) [reply docs/getting-started/quickstart.md:101]: Removed.

File: docs/getting-started/quickstart.md (hunk 5 above)
Comment (seanmooretechwriter) [reply docs/getting-started/quickstart.md:122]: Removed.

File: docs/getting-started/quickstart.md (hunk 6 above)
Comment (seanmooretechwriter) [reply docs/getting-started/quickstart.md:152]: Updated.

File: docs/getting-started/quickstart.md (hunk 3 above)
Comment (maoueh) [reply docs/getting-started/quickstart.md:99]: Ok interesting, could you link me some of them, going to take a look, maybe I'm wrong :)

File: docs/getting-started/quickstart.md (hunk 5 above)
Comment (maoueh) [reply docs/getting-started/quickstart.md:122]: Can you test it though just to be sure it still works without.

File: docs/getting-started/quickstart.md (hunk 9 above)
Comment (seanmooretechwriter) [reply docs/getting-started/quickstart.md:196]: Updated.

**Comment by seanmooretechwriter** [comment]
All changes except those relating to updating the source code for the eth tutorial are complete.

File: docs/getting-started/quickstart.md (hunk 7 above)
Comment (seanmooretechwriter) [reply docs/getting-started/quickstart.md:146]: Code updated in tutorial and pushed to its repo.

File: docs/getting-started/quickstart.md (hunk 8 above)
Comment (seanmooretechwriter) [reply docs/getting-started/quickstart.md]: Updated in tutorial repo and in the code example in this Quickstart.

**Comment by seanmooretechwriter** [comment]
Content ready for review. All changes from last review should be resolved including updates to the accompanying substreams ETH tutorial.

File: docs/getting-started/quickstart.md (hunk 5 above)
Comment (seanmooretechwriter) [reply docs/getting-started/quickstart.md]: Yes. Tested when I updated the code example. Appears to be fully operational.

File: docs/getting-started/quickstart.md (hunk 3 above)
Comment (seanmooretechwriter) [reply docs/getting-started/quickstart.md]: Ha! It looks like where I had read this was wrong! I just found the manifest reference in the Rust docs! My bad! I'll have to find where I read this. I know I did, I researched what to call it when writing this a while back.

Very interesting. Sorry for any confusion!!

https://doc.rust-lang.org/cargo/reference/manifest.html

**Pull request merged**